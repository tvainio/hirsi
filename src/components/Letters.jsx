import React from "react";
import PropTypes from "prop-types";
import { Panel, Label } from "react-bootstrap";

function Letters(props) {

  return (
    <div>
      <Panel>
        <Panel.Body>
          {props.letters.map(letter=> <Label>{letter}</Label>)}
        </Panel.Body>
      </Panel>
    </div>
  );

}

Letters.propTypes = {
  letters: PropTypes.array.isRequired
};

export default Letters;
