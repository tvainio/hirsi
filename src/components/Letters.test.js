import React from "react";
import { shallow } from "enzyme";
import Letters from "./Letters";
import renderer from "react-test-renderer";


it("only shows correctly guessed letters", () => {
  const letters = ["A", "B"];
  const word = "TABLET";
  const wrapper = shallow(<Letters usedLetters={letters} word={word} />);
  expect(wrapper.contains("T")).toBe(false);
  expect(wrapper.contains("A")).toBe(true);
  expect(wrapper.contains("B")).toBe(true);
  expect(wrapper.contains("L")).toBe(false);
  expect(wrapper.contains("E")).toBe(false);
  expect(wrapper.contains("T")).toBe(false);
});
