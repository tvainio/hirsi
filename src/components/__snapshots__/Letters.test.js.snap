// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body"
    >
      Here be the letters...
    </div>
  </div>
</div>
`;
